Source/Editor/Surface/VisjectSurfaceWindow.cs

```diff
@@ -775,7 +775,34 @@ private void EditAttributesParameter(int index, Control label)
 
         private void DeleteParameter(int index)
         {
+            bool displayWarning = Editor.Instance.Options.Options.Interface.WarnOnDeletingUsedVisjectParameter;
+
             var window = (IVisjectSurfaceWindow)Values[0];
+            SurfaceParameter param = window.VisjectSurface.Parameters[index];
+
+            int connectedParameterNodeCount = 0;
+
+            List<SurfaceNode> nodes = window.VisjectSurface.Nodes;
+
+            for (int i = 0; i < window.VisjectSurface.Nodes.Count; i++)
```

Comment:
I would simplify this and just make `bool isParamUsed = window.VisjectSurface.IsParamreferenced(param )`, then ignore plural version of message.

Reply:
Or even better would be to use `VisjectSurfaceContext` to check any nested graphs: eg. if Anim Graph has state machine and that param is used in nested graph but I can take care of that - it will require a little bit more refactor.

Reply:
It'd be very cool if you could implement that.

About dropping the plural version:
My thought was, that if a parameter is being used multiple times in a graph but only one *Get {ParamName}* node is visible, the user may not know that the parameter is being used in an important way a second time.

So then the "being used x times" makes them go "Ohh, wait, I thought this parameter is only being used in the place(s) I'm currently seeing, I should check the other use case(s)!":

*Used2 can be seen one time (red box), but is actually being used a second time (red arrow), but the user can't see that. Granted, this is a very simple node graph, but imagine a more complex one with a lot of nodes*
![image](https://github.com/user-attachments/assets/44d941c6-20df-45b2-89ef-8606482edd37)